src/panel/machine.ts

```diff
@@ -134,6 +136,38 @@ const machine: MachineType = Machine<MachineContext, MachineSchema, MachineEvent
                   };
                 }),
               },
+              SAVE: {
+                internal: true,
+                target: 'idle',
+                // Only allow saving when there are results
+                cond: (context): boolean => {
+                  return context.current.results != null;
+                },
+                actions: assign((context) => {
+                  return {
+                    ...context,
+                    message: 'Result saved',
+                  };
+                }),
+              },
+              LOAD_FROM_FILE: {
```

Comment:
We are effectively re-implementing the `PIN` event right? *Maybe* the `PIN` event could include a `message` value so that we could simply reuse the `PIN` event rather than created another event just so we can have a different message

Reply:
Open to your thoughts though

Reply:
This may have been an artefact of how I was wrapping the way the state machine worked and so it was seperated. But yeah more or less, that would make sense ya.